domain/src/main/java/org/oppia/domain/oppialogger/loguploader/LogUploadWorker.kt

```diff
@@ -0,0 +1,112 @@
+package org.oppia.domain.oppialogger.loguploader
+
+import android.content.Context
+import androidx.work.Worker
+import androidx.work.WorkerParameters
+import org.oppia.domain.oppialogger.analytics.AnalyticsController
+import org.oppia.domain.oppialogger.exceptions.ExceptionsController
+import org.oppia.domain.oppialogger.exceptions.toException
+import org.oppia.util.data.DataProviders
+import org.oppia.util.logging.ConsoleLogger
+import org.oppia.util.logging.EventLogger
+import org.oppia.util.logging.ExceptionLogger
+import javax.inject.Inject
+import javax.inject.Singleton
+
+const val LOG_UPLOAD_WORKER = "log_upload_worker"
+
+/** [Worker] class that extracts log reports from the cache store and logs them to the remote service. */
+@Singleton
+class LogUploadWorker @Inject constructor(
+  context: Context,
+  params: WorkerParameters,
+  private val analyticsController: AnalyticsController,
+  private val exceptionsController: ExceptionsController,
+  private val dataProviders: DataProviders,
+  private val exceptionLogger: ExceptionLogger,
+  private val eventLogger: EventLogger,
+  private val consoleLogger: ConsoleLogger
+) : Worker(context, params) {
+
+  enum class WorkerCase {
+    EVENT_WORKER,
+    EXCEPTION_WORKER
+  }
+
+  companion object {
+    const val WORKER_CASE_KEY = "worker_case_key"
+  }
+
+  override fun doWork(): Result {
+    return when (inputData.getString(WORKER_CASE_KEY)) {
+      WorkerCase.EVENT_WORKER.toString() -> {
+        eventWork()
+      }
+      WorkerCase.EXCEPTION_WORKER.toString() -> {
+        exceptionWork()
+      }
+      else -> Result.failure()
+    }
+  }
+
+  /** Extracts exception logs from the cache store and logs them to the remote service. */
+  private fun exceptionWork(): Result {
+    return try {
+      val exceptionStore =
+        dataProviders.convertToLiveData(exceptionsController.getExceptionLogStore())
+      val exceptionLogs = exceptionStore.value?.getOrThrow()?.exceptionLogList
+      exceptionLogs?.let {
+        for (exceptionLog in it) {
+          exceptionLogger.logException(exceptionLog.toException())
+          it.remove(exceptionLog)
+        }
+      }
+      Result.success()
+    } catch (e: Exception) {
+      consoleLogger.e(LOG_UPLOAD_WORKER, e.toString(), e)
+      Result.failure()
+    }
+  }
+
+  /** Extracts event logs from the cache store and logs them to the remote service. */
+  private fun eventWork(): Result {
+    return try {
+      val eventStore = analyticsController.getEventLogs()
+      val eventLogs = eventStore.value?.getOrThrow()?.eventLogList
```

Comment:
Is this being called on the main thread? If not, we shouldn’t be interacting directly with LiveData like this.

Reply:
Okay, was the way of interaction correct in case of the exceptions ? I've modified this to make the two implementations similar. PTAL.

Reply:
Sorry, this wasn't a comment about consistency. If this function is not being called on the main thread, we cannot retrieve LiveData's value: it's designed to be used only on the main thread. How does AndroidX's WorkManager behave when doWork() is called?

Reply:
Changed the implementation to a coroutine worker and handled cache retrieval via the original controller. PTAL.

Reply:
Looks good--thanks!